src/main/scala/resource/ExtractableManagedResource.scala

```diff
@@ -46,5 +46,7 @@ trait ExtractableManagedResource[+A] extends ManagedResource[A] {
    *        An either where the left hand side is the currently contained resource unless exceptions, in which case
    *        the right hand side will contain the sequence of throwable encountered.
    */
-  def either: Either[Seq[Throwable], A]
+  def either: ExtractedEither[Seq[Throwable], A]
 }
+
+case class ExtractedEither[+A, +B](either: Either[A, B])
```

Comment:
As far as I can see there is no way to fix the bug without introducing either a container of a sub-type of Either. Introducing a container is significantly less effort than a subclass. and placing an implicit conversion in the package object (https://github.com/adamretter/scala-arm/commit/c7e31438b551f34536f1962ebcb1c5bfeb4b6148#diff-c1a4ce80b3e56af8d61bb72ceb8a46b7R75) means that no one using the API will be distrurbed as the `ExtractedEither` will always be unboxed implicitly for them to an `Either`.

To be honest having done a lot of work functional work with Java 8 and `try-with-resources` (which was actually introduced in Java 7), the *suppressed exception* mechanism is just terrible and can lead to very hard to debug problems. Especially if you consider that `try-with-resources` and `managed` are not just for opening and closing IO Streams, but for any object which implements `close`. I would be very disappointed if scala-arm made the same mistakes as Java's `try-with-resources`, you already have a much better mechanism.